### PR TITLE
Add CardRadioGroup and use it for new tier and deductible selectors

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Create new `CardRadioGroup` component as a base for tier and deductible selectors
- Implement first version of new selectors

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4484?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4484** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4484?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4482** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4482?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4481** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4481?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4479** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4479?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4478** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4478?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/features/priceCalculator/DeductibleSelectorV2.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        datadogLogs.logger.warn(`Offer ${offer.id} has no deductible`)
+      }
+    })
+
+    return levels
+  }, [offers])
+
+  return (
+    <>
+      <CardRadioGroup.Root value={selectedOffer.id} onValueChange={onValueChange}>
```
Comment (alebedev) [inline apps/store/src/features/priceCalculator/DeductibleSelectorV2.tsx:52]: There's probably `ProductOfferSelector` component hidden here, we don't want to copy styling between tiers and deductibles

**Review by gustaveen — APPROVED** [review]
🙌

**Comment by alebedev** [comment]
### Merge activity

* **Jul 16, 3:05 AM EDT**: @alebedev merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4484).

**Pull request merged**